scripts/anvil_tools/set_up_anvil_workspaces.py

```diff
@@ -0,0 +1,309 @@
+            else:
+                print("Not a valid option. Choose: Y/N")
+        if update_existing_ws.upper() == "N":
+            return(f"{project}/{workspace_name} already existed. User selected not to overwrite. Try again with unique workspace name.")
+        elif update_existing_ws.upper() == "Y":
+            return ws_exists_status_code  # 200
+    else:
+        return ws_exists_response.json()
+
+
+def make_create_workspace_request(workspace_name, auth_domain_name, project="anvil-datastorage"):
+    """Make the json request to pass into create_workspace()."""
+
+    # initialize empty dictionary
+    create_ws_request = {}
+
+    create_ws_request["namespace"] = project
+    create_ws_request["name"] = workspace_name
+    create_ws_request["authorizationDomain"] = [{"membersGroupName": f'{auth_domain_name}'}]
+    create_ws_request["attributes"] = {}
+    # TODO: set noWorkspaceOwner = True for data delivery workspaces - picard svc is the only owner
+    create_ws_request["noWorkspaceOwner"] = False
+
+    return create_ws_request
+
+
+def add_member_to_auth_domain(auth_domain_name, email, permission):
+    """Add group with given permissions to authorization domain."""
+
+    # request URL for addUserToGroup
+    uri = f"https://api.firecloud.org/api/groups/{auth_domain_name}/{permission}/{email}"
+
+    # Get access token and and add to headers for requests.
+    # -H  "accept: */*" -H  "Authorization: Bearer [token]"
+    headers = {"Authorization": "Bearer " + get_access_token(), "accept": "*/*"}
+
+    # capture response from API and parse out status code
+    response = requests.put(uri, headers=headers)
+    status_code = response.status_code
+
+    # print success or fail message based on status code
+    if status_code == 204:
+        print(f"Successfully updated Authorization Domain, {auth_domain_name}, with group: {email}.")
+        return status_code
+
+    else:
+        print(f"WARNING: Failed to update Authorization Domain, {auth_domain_name}, with group: {email}.")
+        print("Check output file for error details.")
+        return response.json()
+
+
+def create_auth_domain(auth_domain_name):
+    """Create the Terra goolge group to be used as auth domain on workspaces."""
+
+    # request URL for createGroup
+    uri = f"https://api.firecloud.org/api/groups/{auth_domain_name}"
+
+    # Get access token and and add to headers for requests.
+    # -H  "accept: application/json" -H  "Authorization: Bearer [token]"
+    headers = {"Authorization": "Bearer " + get_access_token(), "accept": "application/json"}
+
+    # capture response from API and parse out status code
+    response = requests.post(uri, headers=headers)
+    status_code = response.status_code
+
+    # print success or fail message based on status code
+    if status_code == 201:
+        print(f"Successfully created Authorization Domain with name: {auth_domain_name}.")
+        return status_code
+    elif status_code in ["403", "409"]:
```

Comment:
this can just be `if status_code in ["403", "409"]:` and can remove the `else` 3 lines down